### ISSUE TITLE
Error while copying data from text input 

### ISSUE THREAD

**Issue by Shivangisingh06** [kivy/kivy#5813]
### Versions

* Python: 3.6.5
* OS:Windows 10 64-bit
* Kivy: v1.10.0
* Kivy installation method: pip as mentioned on Kivy installation guide 

### Description
I have created a desktop GUI using kivy and whenever I am trying to copy the data from text input it gives me below mentioned error: 
"ValueError: embedded null character" 

### Code and Logs

```python
[INFO   ] [Logger      ] Record log in C:\Users\ssingh\.kivy\logs\kivy_18-06-18_13.txt
[INFO   ] [Kivy        ] v1.10.0
[INFO   ] [Python      ] v3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 17:00:18) [MSC v.1900 64 bit (AMD64)]
[INFO   ] [Factory     ] 194 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_sdl2, img_pil, img_gif (img_ffpyplayer ignored)
[INFO   ] [Text        ] Provider: sdl2
[INFO   ] [OSC         ] using <thread> for socket
[INFO   ] [Window      ] Provider: sdl2
[INFO   ] [GL          ] Using the "OpenGL" graphics system
[INFO   ] [GL          ] GLEW initialization succeeded
[INFO   ] [GL          ] Backend used <glew>
[INFO   ] [GL          ] OpenGL version <b'4.6.0 NVIDIA 391.03'>
[INFO   ] [GL          ] OpenGL vendor <b'NVIDIA Corporation'>
[INFO   ] [GL          ] OpenGL renderer <b'Quadro P1000/PCIe/SSE2'>
[INFO   ] [GL          ] OpenGL parsed version: 4, 6
[INFO   ] [GL          ] Shading version <b'4.60 NVIDIA'>
[INFO   ] [GL          ] Texture max size <32768>
[INFO   ] [GL          ] Texture max units <32>
[INFO   ] [Window      ] auto add sdl2 input provider
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO   ] [GL          ] NPOT texture support is available
[INFO   ] [Base        ] Start application main loop
[WARNING] [ModalView   ] you can only open once.
[INFO   ] [Clipboard   ] Provider: winctypes
[INFO   ] [Base        ] Leaving application in progress...
 Traceback (most recent call last):
   File "C:\Users\ssingh\source\repos\KivyGuiChanges\KivyGui\main.py", line 152, in <module>
     MainApp().run()
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\app.py", line 828, in run
     runTouchApp()
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\base.py", line 504, in runTouchApp
     EventLoop.window.mainloop()
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\core\window\window_sdl2.py", line 663, in mainloop
     self._mainloop()
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\core\window\window_sdl2.py", line 405, in _mainloop
     EventLoop.idle()
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\base.py", line 342, in idle
     self.dispatch_input()
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\base.py", line 327, in dispatch_input
     post_dispatch_input(*pop(0))
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\base.py", line 233, in post_dispatch_input
     listener.dispatch('on_motion', etype, me)
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:8191)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\core\window\__init__.py", line 1192, in on_motion
     self.dispatch('on_touch_up', me)
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:8191)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\core\window\__init__.py", line 1228, in on_touch_up
     if w.dispatch('on_touch_up', touch):
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:8191)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\uix\textinput.py", line 328, in on_touch_up
     return super(TextInputCutCopyPaste, self).on_touch_up(touch)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\uix\widget.py", line 479, in on_touch_up
     if child.dispatch('on_touch_up', touch):
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:8191)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\uix\widget.py", line 479, in on_touch_up
     if child.dispatch('on_touch_up', touch):
   File "kivy\_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:8191)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\uix\behaviors\button.py", line 179, in on_touch_up
     self.dispatch('on_release')
   File "kivy\_event.pyx", line 714, in kivy._event.EventDispatcher.dispatch (kivy\_event.c:8146)
   File "kivy\_event.pyx", line 1225, in kivy._event.EventObservers.dispatch (kivy\_event.c:14035)
   File "kivy\_event.pyx", line 1109, in kivy._event.EventObservers._dispatch (kivy\_event.c:12816)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\lang\builder.py", line 64, in custom_callback
     exec(__kvlang__.co_value, idmap)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\data\style.kv", line 209, in <module>
     on_release: root.do('copy')
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\uix\textinput.py", line 378, in do
     textinput.copy()
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\uix\textinput.py", line 1712, in copy
     return Clipboard.copy(self.selection_text)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\core\clipboard\__init__.py", line 73, in copy
     self._copy(data)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\core\clipboard\__init__.py", line 87, in _copy
     self.put(data, self._clip_mime_type)
   File "C:\Users\ssingh\AppData\Local\Programs\Python\Python36\lib\site-packages\kivy\core\clipboard\clipboard_winctypes.py", line 55, in put
     msvcrt.wcscpy_s(c_wchar_p(hCd), len(text), c_wchar_p(text))
 ValueError: embedded null character
Press any key to continue . . .

**Comment by dolang** [comment]
This has already been reported in #5569 and fixed with #5579.

The next version should be released shortly and I'd suggest you update to that as soon as it's ready.

If you absolutely need a fix for 1.10.0, you can force Kivy to use the SDL2 clipboard provider instead. You need to [set the environment variable](https://kivy.org/docs/guide/environment.html) `KIVY_CLIPBOARD` to `sdl2`.

You can also put this at the very top of your application:
```py
import os
os.environ['KIVY_CLIPBOARD'] = 'sdl2'
```

**Comment by Shivangisingh06** [comment]
dolang Thankyou
